python-spec/requirements-py3.10.txt

```diff
@@ -2,7 +2,8 @@ anndata==0.8.0
 attrs==22.2.0
 h5py==3.7.0
 natsort==8.2.0
-numpy==1.24.1
+numba==0.56.4
```

Comment:
I'm curious why `==` down to the micro -- my general experience is this is "overpinning" but clearly not the case here

Not specific to this PR; I see other deps are that way too ...

Anyway, just seeking to learn -- cc @thetorpedodog

Reply:
You will have to ask @thetorpedodog 

My guess is that he wants to control the deps for testing purposes.  But either way, I did not introduce this pattern with this PR.

Reply:
> My question here is not specific to this PR; I see other deps are that way too ...
> 
> Anyway, I'm just seeking to learn

> I did not introduce this pattern with this PR.

We are in 100% agreement here